src/Discovery/Discovery.tsx

```diff
@@ -0,0 +1,552 @@
+            setModalData(record);
+          },
+          onKeyPress: () => {
+            setModalVisible(true);
+            setModalData(record);
+          },
+        })}
+        dataSource={visibleResources}
+        expandable={config.study_preview_field && ({
+          // expand all rows
+          expandedRowKeys: visibleResources.map(r => r[config.minimal_field_mapping.uid]),
+          expandedRowRender: (record) => {
+            const studyPreviewText = record[config.study_preview_field.field];
+            const renderValue = (value: string | undefined): React.ReactNode => {
+              if (!value) {
+                if (config.study_preview_field.include_if_not_available) {
+                  return config.study_preview_field.value_if_not_available;
+                }
+              }
+              if (searchTerm) {
+                // get index of searchTerm match
+                const matchIndex = value.toLowerCase().indexOf(searchTerm.toLowerCase());
+                if (matchIndex === -1) {
+                  // if searchterm doesn't match this record, don't highlight anything
+                  return value;
+                }
+                // Scroll the text to the search term and highlight the search term.
+                let start = matchIndex - 100;
+                if (start < 0) {
+                  start = 0;
+                }
+                return (<React.Fragment>
+                  { start > 0 && '...' }
+                  {value.slice(start, matchIndex)}
+                  <span className='matched'>{value.slice(matchIndex, matchIndex + searchTerm.length)}</span>
+                  {value.slice(matchIndex + searchTerm.length)}
+                </React.Fragment>
+                );
+              }
+              return value;
+            };
+            return (
+              <div
+                className='discovery-table__expanded-row-content'
+                role='button'
+                tabIndex={0}
+                onClick={() => {
+                  setModalData(record);
+                  setModalVisible(true);
+                }}
+              >
+                {renderValue(studyPreviewText)}
+              </div>
+            );
+          },
+          expandedRowClassName: () => 'discovery-table__expanded-row',
+          expandIconColumnIndex: -1, // don't render expand icon
+        })}
+      />
+    </div>
+    <Modal
+      className='discovery-modal'
+      visible={modalVisible}
+      onOk={() => setModalVisible(false)}
+      onCancel={() => setModalVisible(false)}
+      width='80vw'
+      footer={false}
+    >
+      <Space direction='vertical' size='large'>
+        { config.study_page_fields.header &&
+          <Space align='baseline'>
+            <h3 className='discovery-modal__header-text'>{modalData[config.study_page_fields.header.field]}</h3>
+            <a href={`/discovery/${modalData[config.minimal_field_mapping.uid]}/`}><LinkOutlined /> Permalink</a>
```

Comment:
Suggestion: wrap this link with `copy` so clicking on it will copy the premalink to clipboard might be a better user experience

Portal is already importing `clipboard-plus` for some other pages, might be able to use that

Reply:
That's a really good idea!! If it's ok with you I'm going to add this to a list of improvements for me to make in the future -- specifically changing this from a link to a button that copies the link.